Common/src/main/java/com/tabnineCommon/lifecycle/BinaryStateSingleton.kt

```diff
@@ -0,0 +1,20 @@
+package com.tabnineCommon.lifecycle
+
+import com.intellij.util.messages.Topic
+import com.tabnineCommon.binary.requests.config.StateResponse
+import com.tabnineCommon.general.TopicBasedState
+import java.util.function.Consumer
+
+class BinaryStateSingleton private constructor() :
```

Comment:
should we consider this singleton to be persisted, til first poling? 
will that make the experience more consistent?

Reply:
I think persisting it can only cause more flakiness and unstable states. Data in the state is considered to be very moment related. Also few IDE can edit it simultaneously.